### PR TITLE
chore: hdfs - remove unnecessary jars and bump dependency versions to fix CVEs

### PR DESCRIPTION
**Pull request by adwk67**
# Description

This PR fixes the following CVEs for snappy-java (bumped from 1.1.8.2 to 1.1.10.4):

CVE-2023-34453
CVE-2023-34454
CVE-2023-43642
CVE-2023-34455

It also strips the image of jars that we do not need (yarn, mapreduce, minicluster modules, plus test/source jars).

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [x] Changes are OpenShift compatible
- [x] All added packages (via microdnf or otherwise) have a comment on why they are added
- [x] Add an entry to the CHANGELOG.md file
- [x] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

File: hadoop/Dockerfile [hunk 1]
```diff
@@ -153,5 +154,20 @@ ENV ASYNC_PROFILER_HOME=/stackable/async-profiler
...
+#  - json-io: this is a transitive dependency pulled in by cedarsoft/java-utils/json-io and is excluded in 3.4.0. See CVE-2023-34610.
+RUN rm -rf /stackable/hadoop/share/hadoop/common/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/hdfs/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/json-io-*.jar && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/hadoop-mapreduce-client-*.jar && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/hadoop-yarn-server*.jar && \
+  find . -name hadoop-minicluster-*.jar -type f -delete && \
+  find . -name hadoop-client-minicluster-*.jar -type f -delete && \
+  find . -name hadoop-*tests.jar -type f -delete
```
Comment (lfrancke) [inline hadoop/Dockerfile:170]: There is the shellcheck warning which actually might make sense here to look at.

File: hadoop/Dockerfile [hunk 2]
```diff
@@ -153,5 +154,20 @@ ENV ASYNC_PROFILER_HOME=/stackable/async-profiler
...
+#  - source code
+#  - mapreduce/yarn binaries that were built as cross-project dependencies
+#  - minicluster (only used for testing) and test .jars
+#  - json-io: this is a transitive dependency pulled in by cedarsoft/java-utils/json-io and is excluded in 3.4.0. See CVE-2023-34610.
+RUN rm -rf /stackable/hadoop/share/hadoop/common/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/hdfs/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/sources/ && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/json-io-*.jar && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/hadoop-mapreduce-client-*.jar && \
+  rm -rf /stackable/hadoop/share/hadoop/tools/lib/hadoop-yarn-server*.jar && \
```
Comment (nightkr) [inline hadoop/Dockerfile:167]: Shouldn't we exclude these from building instead?

Comment (adwk67) [reply hadoop/Dockerfile:167]: We exclude the modules from the final build, but all the jars are built regardless of the `mvn .... -pl` directive, due to cross-dependencies. Hence the second step to strip them here.

Comment (fhennig) [reply hadoop/Dockerfile:167]: it looks like `-pl '!hadoop-tools/hadoop-pipes,!hadoop-yarn-project,!hadoop-mapreduce-project,!hadoop-minicluster'` should exclude at least some of these, maybe it's not working? maybe they cannot be excluded? A comment would be good in that case, explaining why they cannot be excluded from the build.

Comment (adwk67) [reply hadoop/Dockerfile:167]: -pl results in there being e.g. no share/hadoop/yarn or share/hadoop/mapreduce folders in the image, although various yarn/MR binaries are still needed in e.g. share/tools. I'll add a comment.

File: hadoop/Dockerfile (hunk 1 above)
Comment (adwk67) [reply hadoop/Dockerfile]: See 878cc24.

**Review by lfrancke — DISMISSED** [review]

File: hadoop/Dockerfile (hunk 2 above)
Comment (adwk67) [reply hadoop/Dockerfile:169]: Comment is hopefully a bit clearer here: 878cc24

**Review by lfrancke — APPROVED** [review]

File: hadoop/Dockerfile (hunk 2 above)
Comment (fhennig) [reply hadoop/Dockerfile:169]: Works for me

**Pull request merged**